tests/components/lektrico/test_config_flow.py

```diff
@@ -0,0 +1,175 @@
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert not result["errors"]
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: MOCKED_DEVICE_IP_ADDRESS,
+        },
+    )
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_user_setup_device_offline(hass: HomeAssistant, mock_device) -> None:
+    """Test manually setting up when device is offline."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert not result["errors"]
+
+    mock_device.side_effect = DeviceConnectionError
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: MOCKED_DEVICE_IP_ADDRESS,
+        },
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {CONF_HOST: "cannot_connect"}
+    assert result["step_id"] == "user"
+
+    mock_device.side_effect = None
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_HOST: MOCKED_DEVICE_IP_ADDRESS,
+        },
+    )
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+
+
+async def test_discovered_zeroconf(
+    hass: HomeAssistant, mock_device, mock_setup_entry
+) -> None:
+    """Test we can setup when discovered from zeroconf."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data=MOCKED_DEVICE_ZEROCONF_DATA,
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] is None
+    assert result.get("step_id") == "confirm"
+
+    result2 = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+    assert result2["type"] is FlowResultType.CREATE_ENTRY
+    assert result2["data"] == {
+        CONF_HOST: MOCKED_DEVICE_IP_ADDRESS,
+        ATTR_SERIAL_NUMBER: MOCKED_DEVICE_SERIAL_NUMBER,
+        CONF_TYPE: MOCKED_DEVICE_TYPE,
+        ATTR_HW_VERSION: MOCKED_DEVICE_BOARD_REV,
+    }
+    assert result2["title"] == f"{MOCKED_DEVICE_TYPE}_{MOCKED_DEVICE_SERIAL_NUMBER}"
+
+    entry = hass.config_entries.async_entries(DOMAIN)[0]
+    zc_data_new_ip = dataclasses.replace(MOCKED_DEVICE_ZEROCONF_DATA)
+    zc_data_new_ip.ip_address = ip_address(MOCKED_DEVICE_IP_ADDRESS)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data=zc_data_new_ip,
+    )
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+    assert entry.data[CONF_HOST] == MOCKED_DEVICE_IP_ADDRESS
```

Comment:
Please test this in a separate test in the same manner as you did the user already_configured test

Reply:
I made the change, thanks.